src/transaction_builder.cpp

```diff
@@ -456,6 +507,9 @@ TransactionBuilderResult TransactionBuilder::Build()
         } else if (tChangeAddr) {
             // tChangeAddr has already been validated.
             AddTransparentOutput(tChangeAddr.value(), change);
+        } else if (firstOrchardSpendAddr.has_value()) {
+            auto ovk = orchardSpendingKeys[0].ToFullViewingKey().ToInternalOutgoingViewingKey();
+            AddOrchardOutput(ovk, firstOrchardSpendAddr.value(), change, std::nullopt);
```

Comment:
Blocking: I don't understand this condition; why would we ever send funds back to the address that we're spending from rather than to `orchardChangeAddr`? I think that we should require the change address always be set explicitly (and we should remove the next two `else if`s in the process.)

Reply:
This plays to my other comment: this section applies only if the caller doesn't set a change address explicitly. If we want to remove this default behavior, we should do so in a standalone PR, because it's going to impact a bunch of Sapling tests.